src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaUrlRestriction.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Validator\Metadata\Property\Restriction;
+
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Url;
+use Symfony\Component\Validator\Constraints\UrlValidator;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class PropertySchemaUrlRestriction implements PropertySchemaRestrictionMetadataInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @param Url $constraint
+     */
+    public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
+    {
+        $pattern = $constraint->relativeProtocol ? str_replace('(%s):', '(?:(%s):)?', UrlValidator::PATTERN) : UrlValidator::PATTERN;
+
+        $pattern = array_map(static function (string $line) {
+            $line = trim($line);
+
+            $commentPos = strrpos($line, '#');
+            if (false !== $commentPos) {
+                $line = trim(substr($line, 0, $commentPos));
+            }
+
+            return $line;
+        }, explode("\n", $pattern));
+
+        $pattern = substr(implode('', $pattern), 1, -4); // strip options and delimiters
+        $pattern = sprintf($pattern, implode('|', $constraint->protocols));
+
+        return ['pattern' => $pattern];
```

Comment:
I'd suggest to use the available `uri` format on string hence:

```
[
  'type' => 'string',
  'format' => 'uri'
]
```

There are many `URL` formats out there and many corresponding formats on the json-schema specification (http://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3). Might be that Url is actually too broad here and it would need the user to specify the format himself?

Using `#[ApiProperty(schema: ['format' => 'uri'])]` should work (needs check).

Reply:
> I don't get the use of a pattern here, json-schema already validates uris. I'd suggest to use the available `uri` format on string hence:
> 
> ```
> [
>   'type' => 'string',
>   'format' => 'uri'
> ]
> ```
> 
> There are many `URL` formats out there and many corresponding formats on the json-schema specification (http://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3). Might be that Url is actually too broad here and it would need the user to specify the format himself?
> 
> Using `#[ApiProperty(schema: ['format' => 'uri'])]` should work (needs check).

I've added uri format to PropertySchemaFormat

Reply:
okay got it sorry I missed this part, you don't need the regex, it's the responsibility of the schema validator to validate an URI not ours in my opinion.

Reply:
Okay will remove that part